### ISSUE TITLE
[World Loading] Newly created singleplayer worlds disconnect me

### ISSUE THREAD

**Issue by KodaKai1508** [sp614x/optifine#5745]
## Description of Issue
I am attempting to create a new survival world. It loads just fine in vanilla, but when I try to load it up in OptiFine 1.17, the game kicks me from the world. The terrain loads, but then the game says I disconnected, acting as if I was kicked from a multiplayer world.

## Steps to Reproduce
1. Create a new singleplayer survival world in 1.17 OptiFine HD U G9 pre18.
2. Attempt to join singleplayer survival world.
3. Game should kick back to Multiplayer screen giving a disconnect error.

## OptiFine Version
OptiFine HD U G9 pre18, Minecraft version 1.17.0

## Installation Method
I used the given installer.

## Log Files/Crash Reports
[2021-06-09-7.log](https://github.com/sp614x/optifine/files/6626817/2021-06-09-7.log)

## F3 Debug Screenshot
Not available, as the world could not be opened

## Prior Testing
- Does this happen in Vanilla without OptiFine?
No, only with OptiFine and only with this version

## Additional Information
I've never had this issue consistently with any other version of OptiFine or Minecraft.

**Comment by Jiingy** [comment]
I have been able to successfully load into single player worlds on 1.17 pre18 optifine.

do you have any additional mods/resource packs on while trying to load the world?

**Comment by KodaKai1508** [comment]
> do you have any additional mods/resource packs on while trying to load the world?
Yeah, but they've never been giving me issues until now. I could try to reinstall them, but it'll take a while lol

**Comment by Jiingy** [comment]
Try turning off all of your mods and resource packs and see if you can load a world

**Comment by KodaKai1508** [comment]
> Try turning off all of your mods and resource packs and see if you can load a world
Unfortunately, no, my game crashed this time. Here's the log: 
[crash-2021-06-10_13.54.39-client.txt](https://github.com/sp614x/optifine/files/6633453/crash-2021-06-10_13.54.39-client.txt)

**Comment by KaiAF** [comment]
Are you using forge?

**Comment by KaiAF** [comment]
weird. This bug has been reported before, and there (to my knowledge) has not been any fixes before. So I am not sure how to help you with this
